AX: Collapse AccessibilityTree into AccessibilityRenderObject

hoffmanjoshua opened the pull request:
#### 31a738359f918e926a58ecb666857ffec66a8490
<pre>
AX: Collapse AccessibilityTree into AccessibilityRenderObject
<a href="https://bugs.webkit.org/show_bug.cgi?id=298505">https://bugs.webkit.org/show_bug.cgi?id=298505</a>
<a href="https://rdar.apple.com/160026190">rdar://160026190</a>

Reviewed by Tyler Wilcock.

Move all of the logic from AccessibilityTree into superclasses (mostly AccessibilityRenderObject).

* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/accessibility/AXObjectCache.cpp:
(WebCore::AXObjectCache::createObjectFromRenderer):
(WebCore::AXObjectCache::createFromNode):
(WebCore::isAccessibilityTree): Deleted.
* Source/WebCore/accessibility/AccessibilityNodeObject.cpp:
(WebCore::AccessibilityNodeObject::isValidTree const):
(WebCore::AccessibilityNodeObject::computeIsIgnored const):
* Source/WebCore/accessibility/AccessibilityNodeObject.h:
* Source/WebCore/accessibility/AccessibilityObject.h:
* Source/WebCore/accessibility/AccessibilityObjectInlines.h:
(WebCore::AccessibilityObject::hasTreeRole const):
* Source/WebCore/accessibility/AccessibilityRenderObject.cpp:
(WebCore::AccessibilityRenderObject::computeIsIgnored const):
(WebCore::AccessibilityRenderObject::determineAccessibilityRole):
* Source/WebCore/accessibility/AccessibilityTree.cpp: Removed.
* Source/WebCore/accessibility/AccessibilityTree.h: Removed.

Canonical link: <a href="https://commits.webkit.org/299674@main">https://commits.webkit.org/299674@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/35cb51e178a60c0adcf99ba85a3fd6a589faa98d

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/119837 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/39530 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/30181 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/126147 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/71910 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/b6db6713-26c6-4c73-befa-8ca832f2c2bb) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/121713 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/40225 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/48107 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/91013 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/60307 "Passed tests") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/cdd3d743-8f6d-4fd1-a367-7069a87c0341) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/122789 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/32118 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/107457 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/71570 "Found 1 new API test failure: WPE/TestWebKitWebContext:/webkit/WebKitWebContext/no-web-process-leak (failure)") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/8a335114-965c-40f6-8eb5-fcbede2dd9db) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/31148 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/25562 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/69793 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/101590 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/25753 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/129083 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/46757 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/35437 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/99623 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/47123 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/103648 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/99468 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25245 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/44909 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/22925 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/43354 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/46619 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/52325 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/46085 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/49434 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/47771 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

twilco commented on Source/WebCore/accessibility/AccessibilityRenderObject.cpp:
```diff
@@ -1129,6 +1129,35 @@ static bool webAreaIsPresentational(RenderObject* renderer)
     return ownerElement && hasPresentationRole(*ownerElement);
 }
 
+bool AccessibilityRenderObject::isTreeValid() const
```
Any reason you moved this method to AccessibilityRenderObject instead of AccessibilityNodeObject? I think the long term vision is that AccessibilityRenderObject doesn't exist, and AccessibilityNodeObject represents something that either has a node, or a renderer, or both (probably we'll want to different name for the class when we get there)

And even more so, the implementation seems entirely node-based

hoffmanjoshua replied:
I had chosen AccessibilityRenderObject instead of Node object since the only usage of `isTreeValid` is within Render objects. But, you are right that this is a node-baed method, so it does make sense to live inside the AccessibilityNodeObject class.

twilco commented on Source/WebCore/accessibility/AccessibilityRenderObject.h:
```diff
@@ -230,6 +230,8 @@ class AccessibilityRenderObject : public AccessibilityNodeObject {
 
     bool canHavePlainText() const;
+
+    bool isTreeValid() const;
```
Would `isValidTree()` make more sense now that this method isn't in the AccessibilityTree class? "isTreeValid" is ambiguous, e.g. the name could mean a method that checks whether the accessibility tree structure is valid or something

hoffmanjoshua replied:
Yeah, that makes sense to me! Will update

hoffmanjoshua committed:
AX: Collapse AccessibilityTree into AccessibilityRenderObject
https://bugs.webkit.org/show_bug.cgi?id=298505
rdar://160026190

Reviewed by Tyler Wilcock.

Move all of the logic from AccessibilityTree into superclasses (mostly AccessibilityRenderObject).

* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/accessibility/AXObjectCache.cpp:
(WebCore::AXObjectCache::createObjectFromRenderer):
(WebCore::AXObjectCache::createFromNode):
(WebCore::isAccessibilityTree): Deleted.
* Source/WebCore/accessibility/AccessibilityNodeObject.cpp:
(WebCore::AccessibilityNodeObject::isValidTree const):
(WebCore::AccessibilityNodeObject::computeIsIgnored const):
* Source/WebCore/accessibility/AccessibilityNodeObject.h:
* Source/WebCore/accessibility/AccessibilityObject.h:
* Source/WebCore/accessibility/AccessibilityObjectInlines.h:
(WebCore::AccessibilityObject::hasTreeRole const):
* Source/WebCore/accessibility/AccessibilityRenderObject.cpp:
(WebCore::AccessibilityRenderObject::computeIsIgnored const):
(WebCore::AccessibilityRenderObject::determineAccessibilityRole):
* Source/WebCore/accessibility/AccessibilityTree.cpp: Removed.
* Source/WebCore/accessibility/AccessibilityTree.h: Removed.

Canonical link: https://commits.webkit.org/299674@main
